fillBackground.py

```diff
@@ -0,0 +1,112 @@
+                            if fgmasks[a][i][j] and fgmasks[b][i][j] == 0 and sa == 0 and sb:
+                                bg[a][i][j][0] = bg[b][i][j][0]
+                                bg[a][i][j][1] = bg[b][i][j][1]
+                                bg[a][i][j][2] = bg[b][i][j][2]
+                                fgmasks[a][i][j] = 0
+                            elif fgmasks[a][i][j] == 0 and fgmasks[b][i][j] and sb == 0 and sa:
+                                bg[b][i][j][0] = bg[a][i][j][0]
+                                bg[b][i][j][1] = bg[a][i][j][1]
+                                bg[b][i][j][2] = bg[a][i][j][2]
+                                fgmasks[b][i][j] = 0
+            for i in range(height):
+                for j in range(width):
+                    #if the pixel is still 0 then we use average of its surrounding pixel to fill it
+                    if fgmasks[a][i][j]:
+                        I = i-1
+                        while I > -1 and fgmasks[a][I][j]:
+                            I-=1
+                        if I > -1:
+                            B+=bg[a][I][j][0]
+                            G+=bg[a][I][j][1]
+                            R+=bg[a][I][j][2]
+                            count+=1
+                        I = i+1
+                        while I < height and fgmasks[a][I][j]:
+                            I+=1
+                        if I < height:
+                            B+=bg[a][I][j][0]
+                            G+=bg[a][I][j][1]
+                            R+=bg[a][I][j][2]
+                            count+=1
+                        J = j-1
+                        while J > -1 and fgmasks[a][i][J]:
+                            J-=1
+                        if J > -1:
+                            B+=bg[a][i][J][0]
+                            G+=bg[a][i][J][1]
+                            R+=bg[a][i][J][2]
+                            count+=1
+                        J = j+1
+                        while J < width and fgmasks[a][i][J]:
+                            J+=1
+                        if J < width:
+                            B+=bg[a][i][J][0]
+                            G+=bg[a][i][J][1]
+                            R+=bg[a][i][J][2]
+                            count+=1
+
+                        if count:
+                            bg[a][i][j][0] = B/count
+                            bg[a][i][j][1] = G/count
+                            bg[a][i][j][2] = R/count
+                        else:
+                            print("maybe go diagnoal direction ???")
+    def fill_background(self, bg, fgmasks, fps):
+        print('Go fill background ...')
+        n, height, width, channel = bg.shape
+        match = matcher()
+        sampleBG = []
+        # F1=H×F2
+        for i in tqdm(range(1, n, 10)):
+            H = match.match(bg[i-1], bg[i], 'right')
+            # print(H)
+            filled = False 
+            for j in range(height):
```

Comment:
I think we could do matrix multiplication instead of for loop here

Reply:
ok, let us rely on your multiplication